In [33]:
import pandas as pd
import numpy as np

In [34]:
import sqlite3
# Connect to the SQLite database
conn = sqlite3.connect('im.db')

# Query the list of tables in the database
query = "SELECT name FROM sqlite_master WHERE type='table';"
tables = pd.read_sql(query, conn)

print(tables)

            name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers


In [35]:
pd.read_sql('''SELECT * FROM movie_basics LIMIT 2''',conn)

movie_id                    primary_title   original_title  start_year  \
0  tt0063540                        Sunghursh        Sunghursh        2013   
1  tt0066787  One Day Before the Rainy Season  Ashad Ka Ek Din        2019   

   runtime_minutes              genres  
0            175.0  Action,Crime,Drama  
1            114.0     Biography,Drama

In [36]:
pd.read_sql('''SELECT * FROM movie_ratings LIMIT 2''',conn)

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559

In [37]:
# joining the movie_basics table and the movie_ratings table

pd.read_sql('''SELECT *
                FROM movie_basics
                JOIN movie_ratings
                USING(movie_id)
                ORDER BY numvotes DESC
                 LIMIT 10''',conn)

movie_id            primary_title           original_title  start_year  \
0  tt1375666                Inception                Inception        2010   
1  tt1345836    The Dark Knight Rises    The Dark Knight Rises        2012   
2  tt0816692             Interstellar             Interstellar        2014   
3  tt1853728         Django Unchained         Django Unchained        2012   
4  tt0848228             The Avengers             The Avengers        2012   
5  tt0993846  The Wolf of Wall Street  The Wolf of Wall Street        2013   
6  tt1130884           Shutter Island           Shutter Island        2010   
7  tt2015381  Guardians of the Galaxy  Guardians of the Galaxy        2014   
8  tt1431045                 Deadpool                 Deadpool        2016   
9  tt1392170         The Hunger Games         The Hunger Games        2012   

   runtime_minutes                   genres  averagerating  numvotes  
0            148.0  Action,Adventure,Sci-Fi            8.8   1841066  
1            164.0          Action,Thriller            8.4   1387769  
2            169.0   Adventure,Drama,Sci-Fi            8.6   1299334  
3            165.0            Drama,Western            8.4   1211405  
4            143.0  Action,Adventure,Sci-Fi            8.1   1183655  
5            180.0    Biography,Crime,Drama            8.2   1035358  
6            138.0         Mystery,Thriller            8.1   1005960  
7            121.0  Action,Adventure,Comedy            8.1    948394  
8            108.0  Action,Adventure,Comedy            8.0    820847  
9            142.0  Action,Adventure,Sci-Fi            7.2    795227

- Movie with the title inceptions received the highest number of votes at 1841066 votes

In [38]:

pd.read_sql('''SELECT COUNT(*)
                FROM movie_basics
                JOIN movie_ratings
                USING(movie_id)
                WHERE averagerating = 10.0
                 LIMIT 10''',conn)

COUNT(*)
0        16

In [39]:
pd.read_sql('''SELECT genres, averagerating
                FROM movie_basics
                JOIN movie_ratings
                USING(movie_id)
               ORDER BY averagerating DESC
                 LIMIT 17''',conn)

genres  averagerating
0           Documentary           10.0
1          Comedy,Drama           10.0
2     Crime,Documentary           10.0
3           Documentary           10.0
4           Documentary           10.0
5                 Drama           10.0
6           Documentary           10.0
7           Documentary           10.0
8           Documentary           10.0
9           Documentary           10.0
10          Documentary           10.0
11          Documentary           10.0
12  Documentary,History           10.0
13     Adventure,Comedy           10.0
14          Documentary           10.0
15          Documentary           10.0
16                Drama            9.9

- Genre documentary has the highest count of the movies with the highest rating therefore a good genre to venture into for a company.
- other genres that has a rating of 10 are: comedy,Drama,crime,History, and Adventure.

In [40]:
pd.read_sql('''SELECT genres, numvotes
                FROM movie_basics
                JOIN movie_ratings
                USING(movie_id)
               ORDER BY numvotes DESC
                 LIMIT 1''',conn)

genres  numvotes
0  Action,Adventure,Sci-Fi   1841066

- genres Action, Adventure and Sci-Fi are the ones with the highest number of votes